addons/account/models/account_payment.py

```diff
@@ -344,7 +344,9 @@ def _compute_payment_method_id(self):
                 available_payment_methods = pay.journal_id.outbound_payment_method_ids
 
             # Select the first available one by default.
-            if available_payment_methods:
+            if pay.payment_method_id in available_payment_methods:
+                continue
+            elif available_payment_methods:
```

Comment:
This is a compute editable field so:
- you can't use `continue` because you need to assign explicitely a value here even `pay.payment_method_id = pay.payment_method_id` to avoid a `CacheMiss`
- Even when creating the payment from the batch, a compute editable field is not call if you provide a value explicitely. In that case, I think the root cause of your issue is somewhere else

Reply:
Note: maybe missing fields in the batch payment view?

Reply:
@smetl If the ```continue```is applied, it means that pay.payment_method_id has a value.
When setting a partner on the batch payment, the function _compute_payment_method_id defined in module account_check_printing is triggered and the first available payment method is set, that's why function _compute_payment_method_id is triggered